lib/core_ext/open3.rb

```diff
@@ -0,0 +1,52 @@
+module Open3
+  # This is a combination of the functionality of Open3.capture3 and
+  # Open3.pipeline_r
+  #
+  # Open3.pipeline_r doesn't support capturing stderr, so this method takes the
+  # setup code found in `Open3#popen3`, and the block from `Open3#capture3`,
+  # and combines them into a single method.
+  #
+  # Should have the same interface as `Open3.capture3`, but supports passing in
+  # pipes.
+  def pipe_capture3(env, cmds, stdin_data: '', binmode: false, **opts)
```

Comment:
Can we define this method in our own classe instead of monkey patching a core ruby class?

Or is there a hidden method that we need to use that I'm not seeing?

Reply:
> Or is there a hidden method that we need to use that I'm not seeing?

`pipeline_run` later on in this method is what we are using, and that is a private method in the `Open3` module.